src/jobservice/replication/transfer.go

```diff
@@ -460,6 +462,14 @@ func (m *ManifestPusher) enter() (string, error) {
 func newRepositoryClient(endpoint string, insecure bool, credential auth.Credential, repository, scopeType, scopeName string,
 	scopeActions ...string) (*registry.Repository, error) {
 
+	domain, err := config.DomainName()
```

Comment:
I think "hostname" is inaccurate as the value is the external URL which Harbor can be reached, the hostname is not the correct value if Harbor is deployed behind a proxy. So I change it to domain name instead.